src/Uno.UI/UI/Xaml/Media/VisualTreeHelper.cs

```diff
@@ -469,7 +469,7 @@ static VisualTreeHelper()
 			var isChildStale = isStale;
 			while (child.MoveNext())
 			{
-				var childResult = SearchDownForTopMostElementAt(posRelToElement, child.Current!, getVisibility, isChildStale);
+				var childResult = SearchDownForTopMostElementAt(posRelToElement, child.Current!, getVisibility, isChildStale, childrenFilter);
```

Comment:
@dr1rrb Do you know why the recursive call didn't originally include the childrenFilter? I wanted to apply the filter to all the recursive calls, and I thought that was the original intent as well, so I changed it. If that's not correct, I could add a new method that does what I want.

Reply:
Yes most probably a bug ... BTW just noticed that this filter is no longer used, if you don't need it we could just remove it.